pika/channel.py

```diff
@@ -261,17 +259,20 @@ def basic_cancel(self, callback=None, consumer_tag='', nowait=False):
 
         self._cancelled.add(consumer_tag)
 
-        self._rpc(spec.Basic.Cancel(consumer_tag=consumer_tag, nowait=nowait),
-                  self._on_cancelok if not nowait else None,
-                  [(spec.Basic.CancelOk, {'consumer_tag': consumer_tag})] if
-                  nowait is False else [])
+        self._rpc(
+            spec.Basic.Cancel(consumer_tag=consumer_tag, nowait=nowait),
+            self._on_cancelok if not nowait else None, [(spec.Basic.CancelOk, {
+                'consumer_tag':
+                consumer_tag
+            })] if nowait is False else [])
 
-    def basic_consume(self, consumer_callback,
+    def basic_consume(self,
                       queue='',
                       no_ack=False,
                       exclusive=False,
                       consumer_tag=None,
-                      arguments=None):
+                      arguments=None,
+                      callback=None):
```

Comment:
`callback=None` in the function signature suggests that `callback` is optional. However, the code and semantics dictate that it's required. This is likely confusing to the API user.